#45 create module model

PhilippeLeopoldie opened the pull request:
- Implemented the Module model, inheriting from BaseModel.

- Added relationships to Course and Activity.

- Integrated the Module entity into the DbContext.

- Extended the ModuleRepository with Get methods to retrieve:
       . A single module by ID.
       . All modules by CourseID.

PhilippeLeopoldie committed:
feat(Module): Implement Module model and inherit from ILinkBase

PhilippeLeopoldie committed:
feat(Module): Implement Module model and inherit from ILinkBase

PhilippeLeopoldie committed:
feat(Module): Add Course and Activity relations to model

PhilippeLeopoldie committed:
feat(Module): Add Course and Activity relations to model

PhilippeLeopoldie committed:
feat(Module): Add DbContext

PhilippeLeopoldie committed:
feat(Module): Add DbContext

PhilippeLeopoldie committed:
feat(Module): Implement Get methods in Module repository

- Added methods in Module repository to retrieve single and multiple Module entities.
- Supports fetching by ID and retrieving all modules by courseID.

PhilippeLeopoldie committed:
feat(Module): Implement Get methods in Module repository

- Added methods in Module repository to retrieve single and multiple Module entities.
- Supports fetching by ID and retrieving all modules by courseID.

PhilippeLeopoldie committed:
feat(Module) : Inherite from BaseModel

PhilippeLeopoldie committed:
Fix merge conflict

ChornaIryna commented on Domain.Models/Domain.Models.csproj:
```diff
@@ -10,4 +10,8 @@
 
+  <ItemGroup>
+    <ProjectReference Include="..\Service.Contracts\Service.Contracts.csproj" />
+  </ItemGroup>
```
"Den kan tas bort – den behövs inte längre.

PhilippeLeopoldie replied:
Tack! Det är borttaget nu.

ChornaIryna commented on Domain.Models/Entities/Module.cs:
```diff
@@ -0,0 +1,21 @@
+    public Course course { get; set; }
+    public int CourseId { get; set; }
+
+    public ICollection<Activity> activities { get; set; }
```
Property skulle vara med stor "A" - Activities.

PhilippeLeopoldie replied:
Självklart! Det är fixat nu.

PhilippeLeopoldie committed:
feat(Module): Remove dependency on Service.Contracts

ChornaIryna commented on Domain.Models/Entities/Module.cs:
```diff
@@ -0,0 +1,21 @@
+    public DateTime Starts { get; set; }
+    public DateTime Ends { get; set; }
+
+    public Course course { get; set; }
```
Course bör vara required. Jag misstänker att den kommer att orsaka problem vid migreringen eftersom den främmande nyckeln `CourseId` är av typen `int`, vilket inte är nullable.

PhilippeLeopoldie replied:
Tack, bra poäng! Det är fixat nu, tack för feedbacken.

PhilippeLeopoldie committed:
Refactor Module class property names and requirements

Updated property names in the Module class to PascalCase
to adhere to C# naming conventions. Changed 'course' to
'Course' and 'activities' to 'Activities'. Marked the
'Course' property as required to enforce its presence
when creating a Module instance.

PhilippeLeopoldie committed:
fix(Module): correct typo in Activities

PhilippeLeopoldie committed:
refactor(Module): move DbSet<Module> declaration up for clarity

PhilippeLeopoldie committed:
feat(module): update solution
